controlplane/pkg/nsm/healer.go

```diff
@@ -0,0 +1,318 @@
+package nsm
+
+import (
+	"context"
+	"fmt"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/nsm"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/registry"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/model"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/serviceregistry"
+	"github.com/sirupsen/logrus"
+	"time"
+)
+
+type networkServiceHealer interface {
+	healDstDown(healID string, connection *model.ClientConnection) bool
+	healDataplaneDown(healID string, connection *model.ClientConnection) bool
+	healDstUpdate(healID string, connection *model.ClientConnection) bool
```

Comment:
Based on discussion with @haiodo this method (and corresponding event also) could be renamed to `healRemoteDataplaneDown`

Reply:
It seems to be reasonable. But I would rather prefer to postpone such "renaming refactoring" to the subsequent PR.

Reply:
Done

Reply:
Let's do in this PR.